mrmustard/lab_dev/circuit_components.py

```diff
@@ -409,21 +410,22 @@ def fock(self, shape: Optional[int | Sequence[int]] = None, batched=False) -> Co
         Returns:
             array: The Fock representation of this component.
         """
+        num_vars = self.representation.ansatz.num_vars
         if isinstance(shape, int):
-            shape = (shape,) * self.representation.ansatz.num_vars
-        auto_shape = self.auto_shape()
-        shape = shape or auto_shape
-        if len(shape) != len(auto_shape):
-            raise ValueError(
-                f"Expected Fock shape of length {len(auto_shape)}, got length {len(shape)}"
-            )
-
+            shape = (shape,) * num_vars
         try:
             As, bs, cs = self.bargmann_triple(batched=True)
             if self.representation.ansatz.polynomial_shape[0] == 0:
+                auto_shape = self.auto_shape()
+                shape = shape or auto_shape
+                if len(shape) != len(auto_shape):
+                    raise ValueError(
+                        f"Expected Fock shape of length {len(auto_shape)}, got length {len(shape)}"
+                    )
                 arrays = [math.hermite_renormalized(A, b, c, shape) for A, b, c in zip(As, bs, cs)]
             elif self.representation.ansatz.polynomial_shape[0] > 0:
-                num_vars = self.representation.ansatz.num_vars
+                if not shape:
+                    shape = (settings.AUTOCUTOFF_MAX_CUTOFF,) * num_vars
```

Comment:
Can you do the same trick as before
```suggestion
                    shape = shape or (settings.AUTOCUTOFF_MAX_CUTOFF,) * num_vars
```